Tests/BaseObjectTests.swift

```diff
@@ -16,181 +18,183 @@ class BaseObjectTests: QuickSpec {
             }
 
             beforeEach {
-                baseObject = BaseObject()
+                baseObject = ConcreteBaseObject()
                 callbackWasCalled = false
             }
 
-            describe("on") {
-                it("Callback should be called on event trigger") {
-                    baseObject.on(eventName, callback: callback)
-                    baseObject.trigger(eventName)
-
-                    expect(callbackWasCalled) == true
+            context("initialization") {
+                it("doesn't initialize EventDispatcher twice") {
+                    expect(baseObject.eventDispatcher) == baseObject.eventDispatcher
                 }
+            }
 
-                it("Callback should receive userInfo on trigger with params") {
-                    var value = "Not Expected"
-                    baseObject.on(eventName) { userInfo in
-                        value = userInfo?["new_value"] as! String
+            context("events") {
+                describe("on") {
+                    it("executes callback function") {
+                        baseObject.on(eventName, callback: callback)
+                        baseObject.trigger(eventName)
+
+                        expect(callbackWasCalled) == true
                     }
 
-                    baseObject.trigger(eventName, userInfo: ["new_value": "Expected"])
+                    it("receives userInfo when triggered with params") {
```

Comment:
This ``it`` seems that is holding a context logic, do you think that is necessary to split into a ``context`` > ``it`` ?